scanner/src/main/kotlin/scanners/fossid/FossIdScanResults.kt

```diff
@@ -146,25 +150,32 @@ internal fun mapSnippetFindings(
                 FossId.SNIPPET_DATA_RELEASE_DATE to it.releaseDate.orEmpty()
             )
 
-            // TODO: FossID doesn't return the line numbers of the match, only the character range. One must use
-            //       another call "getMatchedLine" to retrieve the matched line numbers. Unfortunately, this is a
-            //       call per snippet which is too expensive. When it is available for a batch of snippets, it can
-            //       be used here.
-            OrtSnippet(
+            var sourceLocation: TextLocation? = null
+            var snippetLocation: TextLocation? = null
+
+            if (it.matchType == MatchType.PARTIAL) {
+                val rawMatchedLines = rawResults.snippetMatchedLines[it.id]
+                val rawMatchedLinesSourceFile = collapseValues(rawMatchedLines?.localFile.orEmpty())
+                val rawMatchedLinesSnippetFile = collapseValues(rawMatchedLines?.mirrorFile.orEmpty())
+
+                sourceLocation = rawMatchedLinesSourceFile.firstOrNull()
+                    ?.let { pair -> TextLocation(it.file, pair.first, pair.second) }
```

Comment:
Use a destructing declaration to give the pair members more meaningful names: `(startLine, endLine) ->`. Same below.